lib/errorHandler.js

```diff
@@ -79,7 +78,16 @@ module.exports = function (config) {
                     body.password = '<omitted>';
                 }
             }
-            logger.error(req.connection.remoteAddress, req.method, req.originalUrl, statusCode, err.stack, err.data, util.inspect(body), util.inspect(req.headers));
+
+            var _logger;
+
+            if (reporter) {
+                _logger = reporter(err, req, res);
+            } else {
+                _logger = logger;
+            }
```

Comment:
Determining which logger should be assigned to  `_logger` is better done in the factory and not in the returned error handler itself.